python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -1007,7 +1104,11 @@ class HourlyPartitionsDefinition(TimeWindowPartitionsDefinition):
             Default is None. Can provide in either a datetime or string format.
         minute_offset (int): Number of minutes past the hour to "split" the partition. Defaults
             to 0.
-        fmt (Optional[str]): The date format to use. Defaults to `%Y-%m-%d`.
+        fmt (Optional[str]): The date format to use. Defaults to `%Y-%m-%d`. Note that if a non-UTC
+            timezone is used, the date format must include a timezone offset to disambiguate between
+            multiple instances of the same time before and after the Fall DST transition. If the
+            format does not contain this offset, the second instance of the ambiguous time partition
+            key will have the UTC offset automatically appended to it.
```

Comment:
doesn't have to block but it might be nice to include something here too https://docs.dagster.io/concepts/partitions-schedules-sensors/schedules#execution-time-and-daylight-savings-time